---
title: "DevKidCC"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{DevKidCC}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(DevKidCC)
library(tidyverse)
library(Seurat)
```

# Introduction

In this vignette I'll run through how to use DevKidCC to classify kidney cells in scRNA-Seq data. I'll then highlight the various visualisation tools that are included in the package.<br>
<br>

## DevKidCC

First we need to load the data and generate a Seurat object. To generate a Seurat object from raw files, follow the tutorials at the [Satija Lab website](https://satijalab.org/seurat/articles/pbmc3k_tutorial.html)<br>
<br>
Here we will load the organoid dataset included with the package, from [Howden et al. 2019](https://doi.org/10.15252/embr.201847483)

```{r}
load("/group/kidn1/Group-Little_MCRI/People/Sean/PhD/R-projects/DevKidCC/data/organoid.rda")
organoid
```
<br>
This is a reduced dataset but it should work for demonstration purposes. This dataset has already been filtered for high quality cells, which should be done prior to any downstream analysis anyway so might as well do it here. What cells are in the dataset will not effect the classification outcome, but poor quality cells will skew dimensional reduction and gene expression analysis.
<br>
<br>
Running the classification tool is a simple one line command:
```{r}
organoid <- DevKidCC(organoid)
```
This runs through each classification stage of the model and outputs all the information including scores and tier outcomes in the metadata.

```{r}
colnames(organoid@meta.data)
organoid@meta.data[1:5, c(23, 25, 27, 28, 32, 53)]
```

Here we can see a comparison between the original identity of the cells `Identity` compared to the scores given for Nephron, Stroma and UrEp identities. The columns `LineageID` and `DKCC` show the assigned identities for the cells at the first (LineageID) and final (DKCC) classification levels.
<br>
<br>
These can now be used to visualise score or classification outcomes on TSNE or UMAP plots, once computed. 
<br>**Hint:** make `cols = myColours()` to replicate the colour scheme used in the paper and custom functions.
<br>

```{r, fig.height=9, fig.width=12}
(DimPlot(organoid, group.by = "LineageID", cols = myColours()) | DimPlot(organoid, group.by = "DKCC", label = T, repel = T, cols = myColours())) /
  (FeaturePlot(organoid, features = "scpred_Nephron", order = T) | FeaturePlot(organoid, features = "PAX2", order = T))
```

## Custom Visualisations

In an effort to make visual analysis of this data easier, we provide some functions for this.
<br>
<br>
***ComponentPlot***
<br>
This function will draw a bar chart showing contribution for annotated identities to a sample. This way you can directly compare two or more samples for the percentage or cell total contributions.
<br>
```{r, fig.width=5, fig.height=5}
ComponentPlot(organoid, identity = "orig.ident", component = "LineageID", show.pct = T, do.label = T, feature = "PAX2")
```
<br>
Personally I prefer flipping the axes
<br>
```{r, fig.width=7, fig.height=3}
ComponentPlot(organoid, identity = "orig.ident", component = "LineageID", show.pct = T, do.label = T, feature = "PAX2") + coord_flip()
```

There are a couple of features you can toggle in this function:<br>
 - `show.pct` will switch between a percentage contribution or total number of cells
 - `do.label` turns the labels inside the bars on or off
 - `show.unassigned` can remove the unassigned cells from the graph 
 - `show.gene.exp` will switch the colours from the identities to a gene expression of the selected feature
 - `feature` changes the gene for the gene expression information if that is toggled on
 
```{r, fig.width=7, fig.height=3}
ComponentPlot(organoid, identity = "orig.ident", component = "LineageID", show.pct = T, do.label = T, feature = "PAX2", show.gene.exp = T) + coord_flip()
```

***DotPlotCompare***
<br>
This function is a modified version of `Seurat::DotPlot()`. I've modified this by allowing you to express the original reference data and if wanted the previously published organoid datasets, as classified using `DevKidCC`. This function can be used to directly compare gene expression profiles between datasets.
<br>

```{r, fig.width=9, fig.height=6}
DotPlotCompare(organoid, features = c("PAX2", "EPCAM", "NPHS1", "COL3A1", "GATA3"), columns = 1, size = "pct.exp")
```

I've tried to include multiple usage methods for this function but the simplest is to call your dataset and the genes you are interested in.
<br>
There are some fine tuning steps you can use if desired:
<br>
Select specific identities. There are key words you can use; "nephron" will select only the nephron segments, or you can select one or more specific identites. You can also switch the grouping from "gene" to "segment" and change the relative size of the dots with "dot.scale".

```{r, fig.width=12, fig.height=6}
DotPlotCompare(organoid, features = c("PAX2", "EPCAM", "NPHS1", "COL3A1", "GATA3"), columns = 1, size = "pct.exp", idents = "nephron") |
  DotPlotCompare(organoid, features = c("PAX2", "EPCAM", "NPHS1", "COL3A1", "GATA3"), columns = 1, 
                 size = "pct.exp", idents = "EPod", show = "segment", dot.scale = 10)
```

<br>
To compare your dataset to the existing organoid database, change `compare.to.organoids`. This function doesn't filter specific organoids so it isn't the easiest on the eye. Hopefully I can add that feature at a later date. Our test organoids are at the top or right (if flipped). Prior to using this call, you will need to download the organoid database. The database is stored as an rda file and can be downloaded at the following link:
https://drive.google.com/file/d/1_6L0EKsYcHq2cS7cRK2NM5G6JCEZClW-/view?usp=sharing


```{r, fig.height=5, fig.width=12}
DotPlotCompare(organoid, features = c("PAX2", "EPCAM", "NPHS1", "COL3A1", "GATA3"), compare.to.organoids = T, 
               columns = 1, size = "pct.exp", idents = "EPod", show = "segment") + coord_flip()

```

***SankeyPlot***
<br>
I find Sankey plots aesthetically pleasing, and can be a good way to visualise the flow of cell classification between identities, samples or just a breakdown of the tiers.
<br>

```{r}
SankeyPlot(organoid, origin = "orig.ident")
```
<br>
There are an number of parameters you can tweak such as the width and height, font size, whether to show the intermediate classification steps with `simple` or change how the colours are assigned using `LinkGroup`, "Var1" is the origin and "Var2" is the end identity.

```{r}
SankeyPlot(organoid, origin = "orig.ident", end = "DKCC", simple = F, width = 500, height = 500, LinkGroup = "Var2", fontsize = 15)


```

***IdentBoxPlot***
<br>
This is a function to plot the assigned identities for multiple samples in a single Seurat object against eachother in a comparative manner. It doesn't provide any statistics however, just a visual representation which can be useful with large dataset of many samples, unlike this example.
<br>
```{r, fig.width=7, fig.height=4}
IdentBoxPlot(organoid, group = "orig.ident", component = "LineageID", show.pct = T, column = 4)
```



